bundles/org.eclipse.emfcloud.modelserver.emf/src/org/eclipse/emfcloud/modelserver/emf/common/EMFFacetConstraints.java

```diff
@@ -0,0 +1,171 @@
+   private List<String> pattern;
+   @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IntegerFilter.class)
+   private Integer totalDigits;
+   @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IntegerFilter.class)
+   private Integer fractionDigits;
+   @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IntegerFilter.class)
+   private Integer length;
+   @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IntegerFilter.class)
+   private Integer minLength;
+   @JsonInclude(value = JsonInclude.Include.CUSTOM, valueFilter = IntegerFilter.class)
+   private Integer maxLength;
+   private String minExclusive;
+   private String maxExclusive;
+   private String minInclusive;
+   private String maxInclusive;
+
+   public Integer getWhiteSpace() { return whiteSpace; }
+
+   public void setWhiteSpace(final Integer whiteSpace) { this.whiteSpace = whiteSpace; }
+
+   public List<String> getEnumeration() { return enumeration; }
+
+   public void setEnumeration(final List<String> enumeration) { this.enumeration = enumeration; }
+
+   public List<String> getPattern() { return pattern; }
+
+   public void setPattern(final List<String> pattern) { this.pattern = pattern; }
+
+   public Integer getTotalDigits() { return totalDigits; }
+
+   public void setTotalDigits(final Integer totalDigits) { this.totalDigits = totalDigits; }
+
+   public Integer getFractionDigits() { return fractionDigits; }
+
+   public void setFractionDigits(final Integer fractionDigits) { this.fractionDigits = fractionDigits; }
+
+   public Integer getLength() { return length; }
+
+   public void setLength(final Integer length) { this.length = length; }
+
+   public Integer getMinLength() { return minLength; }
+
+   public void setMinLength(final Integer minLength) { this.minLength = minLength; }
+
+   public Integer getMaxLength() { return maxLength; }
+
+   public void setMaxLength(final Integer maxLength) { this.maxLength = maxLength; }
+
+   public String getMinExclusive() { return minExclusive; }
+
+   public void setMinExclusive(final String minExclusive) { this.minExclusive = minExclusive; }
+
+   public String getMaxExclusive() { return maxExclusive; }
+
+   public void setMaxExclusive(final String maxExclusive) { this.maxExclusive = maxExclusive; }
+
+   public String getMinInclusive() { return minInclusive; }
+
+   public void setMinInclusive(final String minInclusive) { this.minInclusive = minInclusive; }
+
+   public String getMaxInclusive() { return maxInclusive; }
+
+   public void setMaxInclusive(final String maxInclusive) { this.maxInclusive = maxInclusive; }
+
+   public EMFFacetConstraints(final Map<String, Object> facetMap) {
+      this.whiteSpace = (Integer) facetMap.getOrDefault(WHITESPACE, null);
+      this.enumeration = (List<String>) facetMap.getOrDefault(ENUMERATION, null);
+      this.pattern = (List<String>) facetMap.getOrDefault(PATTERN, null);
```

Comment:
```suggestion
   @SuppressWarnings("unchecked")
   public EMFFacetConstraints(final Map<String, Object> facetMap) {
      this.whiteSpace = (Integer) facetMap.getOrDefault(WHITESPACE, null);
      this.enumeration = (List<String>) facetMap.getOrDefault(ENUMERATION, null);
      this.pattern = (List<String>) facetMap.getOrDefault(PATTERN, null);
```

Please void warning here